Source/WebCore/Modules/push-api/PushDatabase.cpp

```diff
@@ -211,6 +211,8 @@ static Expected<UniqueRef<SQLiteDatabase>, ShouldDeleteAndRetry> openAndMigrateD
     }
 
     if (version < currentPushDatabaseVersion) {
+        FileSystem::setExcludedFromBackup(FileSystem::parentPath(path), true);
```

Comment:
Shouldn't we call this function as well in case path does not exist before we call db->open and db->open is successful?

Reply:
I'm pretty sure you have to call this function on a path that actually exists on the system, because I think the way it works is by call setxattr on the path. So we have to call it after `FileSystem::makeAllDirectories`.

I guess I should also say that this PR isn't strictly necessary, because there's another part of DB initialization that compares the persisted device token to the current device token and wipes the DB on mismatch. This would happen after a backup/restore. But it's better to just avoid backing up the DB entirely if possible.

Reply:
What do we expect to happen if the backup is restored to the same device (device is reset)? Do we expect push data to lose?